### PR TITLE
Create YAML model files

### PR DESCRIPTION
**Pull request by lovincyrus**
INSERT DESCRIPTION HERE

Fixes https://linear.app/rilldata/issue/ENG-856/add-a-create-model-menu-item-for-connector-files

**Checklist:**
- [ ] Covered by tests
- [ ] Ran it and it works as intended
- [ ] Reviewed the diff before requesting a review
- [ ] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [ ] I'm proud of this work!

### CODE REVIEW

**Comment by lovincyrus** [comment]
Demo with dev limit

https://github.com/user-attachments/assets/46c48e92-1695-46ed-bf69-db9dd4eaa70f

**Comment by lovincyrus** [comment]
Verified that we can create YAML model files for various non-OLAP connectors except for redshift. Logged a ticket [here](https://linear.app/rilldata/issue/ENG-858/unable-to-view-redshift-table-model-properly)

**Comment by lovincyrus** [comment]
https://linear.app/rilldata/issue/ENG-860/humanize-redshift-error-when-output-location-and-role-arn-are-missing

Demo 

https://github.com/user-attachments/assets/c0dc68ed-d22b-45d7-b02a-9c2401b99214

File: web-common/src/features/entity-management/error-utils.ts
Comment (ericpgreen2) [inline web-common/src/features/entity-management/error-utils.ts:1]: Since this special error handling is only used for Models, it could be nice to move this file to the `features/models` directory. I think of the `features/entity-management` directory as a place for code that relates to _all_ Rill resource types.

File: web-common/src/features/connectors/selectors.ts [hunk 1]
```diff
@@ -56,15 +68,21 @@ function createModelingSupportQueryOptions(
...
+        return isSqlModelingSupported(spec);
+      } else {
+        // YAML modeling is supported for connectors that support connector type specification:
+        // These connectors use YAML files with "connector: postgres" syntax
+        // - Postgres, MySQL, SQL Server, etc. (external databases)
+        // - NOT DuckDB (uses SQL models)
+        // - NOT provisioned connectors (uses SQL models)
+        // - NOT read-write mode connectors (uses SQL models)
+        return !isSqlModelingSupported(spec);
+      }
```
Comment (ericpgreen2) [inline web-common/src/features/connectors/selectors.ts]: This is confusing because, when modeling is supported, the user can use either SQL or YAML – whatever they prefer. (However, we're opinionated and think that SQL should be used for basic models and YAML should be used for [advanced models](https://docs.rilldata.com/build/advanced-models/).)

So, to avoid confusion, it feels like function and variable names need to be changed.

File: web-common/src/features/connectors/olap/olap-config.ts [hunk 2]
```diff
@@ -53,6 +53,12 @@ export function makeSufficientlyQualifiedTableName(
     case "pinot":
       // TODO
       return table;
+    case "mysql":
+      // MySQL uses database.table format (no schema concept like PostgreSQL)
+      if (database && database !== "default") {
+        return `${database}.${table}`;
+      }
+      return table;
```
Comment (ericpgreen2) [inline web-common/src/features/connectors/olap/olap-config.ts:61]: Given this code now applies to non-OLAP connectors, it seems this file should be renamed and moved out of the `connectors/olap` directory.

File: web-common/src/features/connectors/explorer/TableMenuItems.svelte [hunk 3]
```diff
@@ -79,7 +107,14 @@
   </NavigationMenuItem>
 {/if}
 
-{#if showGenerateMetricsAndDashboard}
+{#if isYamlModelingSupported}
```
Comment (ericpgreen2) [inline web-common/src/features/connectors/explorer/TableMenuItems.svelte]: Here's a case where this variable name is confusing. When modeling is supported, both SQL and YAML are supported.

File: web-common/src/features/connectors/explorer/TableEntry.svelte [hunk 4]
```diff
@@ -41,6 +42,10 @@
   );
   $: isModelingSupported = $isModelingSupportedForConnector.data;
 
+  $: isYamlModelingSupportedForConnector =
+    useIsYamlModelingSupportedForConnector(runtimeInstanceId, connector);
+  $: isYamlModelingSupported = $isYamlModelingSupportedForConnector.data;
```
Comment (ericpgreen2) [inline web-common/src/features/connectors/explorer/TableEntry.svelte]: Here too– variable names need to change.

**Review by ericpgreen2 — DISMISSED** [review]
Code is clean! Left a few comments.

File: web-common/src/features/connectors/explorer/TableMenuItems.svelte (hunk 3 above)
Comment (lovincyrus) [reply web-common/src/features/connectors/explorer/TableMenuItems.svelte]: ~~Do we only want users to create YAML models going forward?~~

File: web-common/src/features/connectors/explorer/TableEntry.svelte (hunk 4 above)
Comment (lovincyrus) [reply web-common/src/features/connectors/explorer/TableEntry.svelte]: Updated to "usePrefers" instead of "useIsSupported"

File: web-common/src/features/connectors/selectors.ts (hunk 1 above)
Comment (lovincyrus) [reply web-common/src/features/connectors/selectors.ts]: https://github.com/rilldata/rill/pull/7716#discussion_r2255144647

File: web-common/src/features/connectors/code-utils.ts [hunk 5]
```diff
@@ -235,3 +256,168 @@ export function replaceOlapConnectorInYAML(
...
+    database,
+    databaseSchema,
+    table,
+  );
+
+  // Use the sufficiently qualified table name directly
+  const selectStatement = `select * from ${sufficientlyQualifiedTableName}`;
+
+  // Determine if dev section should be included based on connector type
+  const shouldIncludeDevSection = driverName !== "redshift";
```
Comment (ericpgreen2) [inline web-common/src/features/connectors/code-utils.ts:306]: Can you add a comment to explain why we're not providing a `dev` section for Redshift? I.e. Redshift does not support `LIMIT` clauses in its `UNLOAD` data exports.

File: web-common/src/features/connectors/selectors.ts [hunk 6]
```diff
@@ -12,16 +12,31 @@ import {
...
 } from "../../runtime-client";
 import { ResourceKind } from "../entity-management/resource-selectors";
 import type { ErrorType } from "@rilldata/web-common/runtime-client/http-client";
 
+// Helper function to determine if a connector prefers SQL-based modeling
+// Connectors that prefer SQL models include:
+// - DuckDB (embedded database with full SQL support)
+// - Provisioned (managed) connectors
+// - Read-write mode connectors
+function prefersSqlBasedModeling(spec: V1ConnectorSpec) {
```
Comment (ericpgreen2) [inline web-common/src/features/connectors/selectors.ts]: Unfortunately this variable name (and the related framing throughout the rest of the PR) is still confusing. All the cases here are cases where modeling is supported, period. It doesn't have anything to do with preferring SQL or YAML.

Comment (ericpgreen2) [reply web-common/src/features/connectors/selectors.ts]: Perhaps the cases you're looking for are:
1. Connector is an OLAP connector that supports modeling (which this function is detecting)
2. Connector is an OLAP connector that doesn't support modeling
3. Connector is not an OLAP connector (or: the connector is a SQL or Warehouse connector)

**Review by ericpgreen2 — DISMISSED** [review]

File: web-common/src/features/connectors/selectors.ts (hunk 6 above)
Comment (lovincyrus) [reply web-common/src/features/connectors/selectors.ts]: https://github.com/rilldata/rill/pull/7716/commits/2d36c4eee39eb7aad3d92eb7bc2ba9fd45085fa5

Comment (lovincyrus) [reply web-common/src/features/connectors/selectors.ts]: connected the dots and made some changes!

File: web-common/src/features/connectors/explorer/TableMenuItems.svelte [hunk 7]
```diff
@@ -70,16 +71,40 @@
...
+{#if isModelingSupported || showGenerateModel}
+  <NavigationMenuItem on:click={handleCreateModelFromTable}>
     <Model slot="icon" />
-    Create new model
+    Create model
   </NavigationMenuItem>
 {/if}
 
-{#if showGenerateMetricsAndDashboard}
+{#if showGenerateMetricsAndDashboard && isModelingSupported}
```
Comment (ericpgreen2) [inline web-common/src/features/connectors/explorer/TableMenuItems.svelte]: This doesn't look like it'll work for Druid & Pinot & self-hosted ClickHouse. For all these cases, we don't support modeling, but we do want the user to be able to create metrics views on top of their OLAP's tables.

**Review by ericpgreen2 — DISMISSED** [review]

File: web-common/src/features/connectors/explorer/TableMenuItems.svelte (hunk 7 above)
Comment (lovincyrus) [reply web-common/src/features/connectors/explorer/TableMenuItems.svelte]: https://github.com/rilldata/rill/pull/7716/commits/c270fb11719aa75899ff5da98194c0342bfe93c7

**Review by ericpgreen2 — APPROVED** [review]
👌
